mongomock/aggregate.py

```diff
@@ -543,6 +544,29 @@ def _handle_type_convertion_operator(self, operator, values):
                 raise TypeError("'%s' type is not supported" % type(parsed))
             return decimal_value
 
+        # Document: https://docs.mongodb.com/manual/reference/operator/aggregation/arrayToObject/
+        if operator == '$arrayToObject':
+            try:
+                parsed = self.parse(values)
+            except KeyError:
+                return None
+
+            if parsed is None:
+                return None
+
+            if not isinstance(parsed, (list, tuple)):
+                raise OperationFailure('$arrayToObject requires an array input, '
```

Comment:
Please break after `OperationFailure(`: that helps with cleaner indentation that does not depend on the length of the class name.